app/Http/Controllers/BookingController.php

```diff
@@ -2,7 +2,87 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Booking;
+use App\Models\Room;
+use App\Services\JsonResponseService;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+
 class BookingController extends Controller
 {
-    //
+    private JsonResponseService $responseService;
+
+    public function __construct(JsonResponseService $responseService)
+    {
+        $this->responseService = $responseService;
+    }
+
+    public function create(Request $request)
+    {
+        $hidden = ['created_at', 'updated_at'];
+        $booking = new Booking();
+
+        $booking->customer = $request->customer;
+        $booking->guests = $request->guests;
+        $booking->start = $request->start;
+        $booking->end = $request->end;
+
+        if ($request->end < $request->start) {
+            return response()->json($this->responseService->getFormat(
+                'Start date must be before the end date'), 400);
+        }
+
+        $clash = Booking::query()->join('booking_room', 'booking_id', '=', 'booking_id')
+            ->where('room_id', $request->room_id)
+            ->where('end', '>=', $request->start)
+            ->where('start', '<=', $request->start)
+            ->exists();
```

Comment:
Your logic here isn't quite working, it's almost there.

To test this, I create a booking that started "2024-04-09" and ended "2024-04-13".

I was not able to create a second booking with the same dates, but if I changed the start in postman to "2024-04-08" I was able to create the booking.

Reply:
I would suggest that you start by writing a new tests that tests this exact senario, you should find that the test fails.

That will hopefully make it easier for you to test the logic as you go.